### PR TITLE
session: api to get current cache size

### LINKED ISSUE #4095: Get cache size
```session``` module has methods for cache clearing, but is it possible to get the size of it?

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes #4095

### CODE REVIEW

**Comment by zcbenz** [comment]
Why not use `CalculateSizeOfAllEntries`?

File: atom/browser/api/atom_api_session.h [hunk 1]
```diff
@@ -38,6 +38,11 @@ class Session: public mate::TrackableObject<Session>,
  public:
   using ResolveProxyCallback = base::Callback<void(std::string)>;
 
+  enum CacheAction {
```
Comment (zcbenz) [inline atom/browser/api/atom_api_session.h]: Should use `enum class` if you don't add prefix for the values.

File: docs/api/session.md [hunk 2]
```diff
@@ -159,6 +159,13 @@ on complete.
 Removes the cookies matching `url` and `name`, `callback` will called with
 `callback()` on complete.
 
+#### `ses.getCacheSize(callback)`
+
+* `callback` Function
+  * `size` Integer
+
+Returns the session's current cache size.
```
Comment (zcbenz) [inline docs/api/session.md]: Should clarify the unit of size.

File: atom/browser/api/atom_api_session.cc [hunk 3]
```diff
@@ -287,10 +304,12 @@ void Session::ResolveProxy(const GURL& url, ResolveProxyCallback callback) {
   new ResolveProxyHelper(browser_context(), url, callback);
 }
 
-void Session::ClearCache(const net::CompletionCallback& callback) {
+template<Session::CacheAction action>
+void Session::GetCache(const net::CompletionCallback& callback) {
```
Comment (zcbenz) [inline atom/browser/api/atom_api_session.cc]: `GetCache` is a bad name here, probably use `DoCacheAction`.

**Comment by deepak1556** [comment]
`CalculateSizeOfAllEntries` is not available yet, have addressed other comments. Thanks!

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**